data/mods/gen2stadium/scripts.ts

```diff
@@ -0,0 +1,561 @@
+				defense = target.getStat(defType, true, true);
+			}
+
+			if (attack >= 256 || defense >= 256) {
+				attack = this.battle.clampIntRange(Math.floor(this.battle.clampIntRange(attack, 1, 999) / 4), 1);
+				defense = this.battle.clampIntRange(Math.floor(this.battle.clampIntRange(defense, 1, 999) / 4), 1);
+			}
+
+			// Self destruct moves halve defense at this point.
+			if (move.selfdestruct && defType === 'def') {
+				defense = this.battle.clampIntRange(Math.floor(defense / 2), 1);
+			}
+
+			// Let's go with the calculation now that we have what we need.
+			// We do it step by step just like the game does.
+			let damage = level * 2;
+			damage = Math.floor(damage / 5);
+			damage += 2;
+			damage *= basePower;
+			damage *= attack;
+			damage = Math.floor(damage / defense);
+			damage = Math.floor(damage / 50);
+			if (isCrit) damage *= 2;
+			damage = Math.floor(this.battle.runEvent('ModifyDamage', attacker, defender, move, damage));
+			damage = this.battle.clampIntRange(damage, 1, 997);
+			damage += 2;
+
+			// Weather modifiers
+			if (
+				(type === 'Water' && this.battle.field.isWeather('raindance')) ||
+				(type === 'Fire' && this.battle.field.isWeather('sunnyday'))
+			) {
+				damage = Math.floor(damage * 1.5);
+			} else if (
+				((type === 'Fire' || move.id === 'solarbeam') && this.battle.field.isWeather('raindance')) ||
+				(type === 'Water' && this.battle.field.isWeather('sunnyday'))
+			) {
+				damage = Math.floor(damage / 2);
+			}
+
+			// STAB damage bonus, the "???" type never gets STAB
+			if (type !== '???' && pokemon.hasType(type)) {
+				damage += Math.floor(damage / 2);
+			}
+
+			// Type effectiveness
+			const totalTypeMod = target.runEffectiveness(move);
+			// Super effective attack
+			if (totalTypeMod > 0) {
+				if (!suppressMessages) this.battle.add('-supereffective', target);
+				damage *= 2;
+				if (totalTypeMod >= 2) {
+					damage *= 2;
+				}
+			}
+			// Resisted attack
+			if (totalTypeMod < 0) {
+				if (!suppressMessages) this.battle.add('-resisted', target);
+				damage = Math.floor(damage / 2);
+				if (totalTypeMod <= -2) {
+					damage = Math.floor(damage / 2);
+				}
+			}
+
+			// Apply random factor is damage is greater than 1, except for Flail and Reversal
+			if (!move.noDamageVariance && damage > 1) {
+				damage *= this.battle.random(217, 256);
+				damage = Math.floor(damage / 255);
+			}
+
+			// If damage is less than 1, we return 1
+			if (basePower && !Math.floor(damage)) {
+				return 1;
+			}
+
+			// We are done, this is the final damage
+			return damage;
+		},
+	},
+	/**
+	 * Stadium 2 ignores stat drops due to status ailments upon boosting the dropped stat.
+	 * For example: if a burned Snorlax uses Curse then it will ignore the attack drop from burn when it is recalculating its attack stat.
+	 * This is why volatiles are added to status conditions, so that we can keep track of whether or not to apply the stat drop from statuses.
```

Comment:
```suggestion
	 * For example: if a burned Snorlax uses Curse then it will ignore the attack drop from
	 * burn when it is recalculating its attack stat. This is why volatiles are added to status
	 * conditions, so that we can keep track of whether or not to apply the stat drop from
	 * statuses.
```